docs/source/deployguide/ca-config.md

```diff
@@ -0,0 +1,423 @@
+- **csr.keyrequest**: Use these values ff you want to customize the crypto [algorithm and key sizes](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/users-guide.html#initializing-the-server).
+- **csr.names**: Specify the values you want to use for the certificate Issuer, visible in the signing certificate.
+- **csr.hosts**: Provide the host names that the server will be running on.
+- **csr.expiry**: Specify when this CA's root certificate expires. The default value `131400h` is 15 years.
+- **csr.pathlength**: If you will have an intermediate CA, set this value to `1` in the root CA. If this is an intermediate CA the value would be `0` unless it will serve as a parent CA for another intermediate CA.
+
+## signing
+
+```
+signing:
+    default:
+      usage:
+        - digital signature
+      expiry: 8760h
+    profiles:
+      ca:
+         usage:
+           - cert sign
+           - crl sign
+         expiry: 43800h
+         caconstraint:
+           isca: true
+           maxpathlen: 0
+           maxpathlenzero: true
+      tls:
+         usage:
+            - signing
+            - key encipherment
+            - server auth
+            - client auth
+            - key agreement
+         expiry: 8760h
+```
+
+The defaults in this section are normally sufficient for a production server. However, you might want to modify the default expiration of the generated enrollment CA and TLS certificates. Note that this is different than the `expiry` specified in the `csr` section for the CA root certificate.  
+
+If this is a TLS CA, it is recommended that you remove the `ca` section from `profiles:` since a TLS CA should only be issuing TLS certificates.
+
+If you plan to have more than one level of intermediate CAs, then you must set `maxpathlen` to greater than `0` in the configuration .yaml file for the root CA. This field represents the maximum number of non-self-issued intermediate certificates that can follow this certificate in a certificate chain. For example, if you plan to have intermediate CAs under this root CA, the `maxpathlen` can be set to `0`. But if you want your intermediate CA to serve as a parent CA to another intermediate CA, then `maxpathlen` should be set to `1`.
+
+To enforce a `maxpathlen` of `0`, you need to also set `maxpathlenzero` to true. If `maxpathlen` is greater than `0`, `maxpathlenzero` should be set to `false`.
+
+## bccsp
+
+```
+bccsp:
+    default: SW
+    sw:
+        hash: SHA2
+        security: 256
+        filekeystore:
+            # The directory used for the software file-based keystore
+            keystore: msp/keystore
+```
+
+If you plan to use a [Hardware Security Module](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/users-guide.html#example) (HSM) you need to configure the `hash` and `security` of the HSM crypto library. When certificates are generated, the private key is generated by and stored in the HSM instead of the `msp/keystore` folder.
```

Comment:
I pulled it from [here ](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/serverconfig.html) where it is missing as well. Is this what I should add?
```default: PKCS11
  pkcs11:
    Library: /etc/hyperledger/fabric/libsofthsm2.so
    Pin: 71811222
    Label: fabric
    hash: SHA2
    security: 256
    Immutable: false

Reply:
I've changed it to this. I like it much better:

The information in this section controls where the private key for the CA is stored. The configuration above causes the private key to be stored on the file system of the CA server in the `msp/keystore` folder. If you plan to use a Hardware Security Module (HSM), the configuration is different. When you [configure HSM for a CA](../hsm.html#using-an-hsm-with-a-fabric-ca), the CA private key is generated by and stored in the HSM instead of the `msp/keystore` folder. An example of the HSM configuration for softHSM would be similar to:

```
bccsp:
  default: PKCS11
  pkcs11:
    Library: /etc/hyperledger/fabric/libsofthsm2.so
    Pin: 71811222
    Label: fabric
    hash: SHA2
    security: 256
    Immutable: false
```